File: 8. ContactsManagerApp/app/src/main/java/com/example/contactsmanagerapp/MainActivity.java
package com.example.contactsmanagerapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.example.contactsmanagerapp.db.Contacts;
import com.example.contactsmanagerapp.db.ContractsRoomDatabase;
import com.google.android.material.textfield.TextInputEditText;
import com.google.android.material.textfield.TextInputLayout;

import java.util.List;

public class MainActivity extends AppCompatActivity {


    TextInputEditText enterNameField, enterEmailField;
    Button button;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        enterNameField = findViewById(R.id.enterNameField);
        enterEmailField = findViewById(R.id.enterEmailField);

        button = findViewById(R.id.button);

        showDataFromDatabase(this);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String enteredName = enterNameField.getText().toString();
                String enteredEmail = enterEmailField.getText().toString();

                insertIntoDatabase(MainActivity.this, enteredName, enteredEmail);

            }
        });


    }

    void insertIntoDatabase(Context context, String enterNameField, String enterEmailField){
        Thread thread  = new Thread(new Runnable() {
            @Override
            public void run() {
                Contacts contact = new Contacts();
                contact.setContact_name(enterNameField);
                contact.setContact_email(enterEmailField);
                ContractsRoomDatabase.getInstance(context).contactsDAO().insertContact(contact);

            }
        });
        thread.start();
    }

    void showDataFromDatabase(Context context){
        Thread thread  = new Thread(new Runnable() {
            @Override
            public void run() {
                List<Contacts> contactsList =  ContractsRoomDatabase.getInstance(context).contactsDAO().getAllContacts();
                Log.i("contactsList",contactsList.toString());

            }
        });
        thread.start();
    }


}

File: __Kotlin Apps/FragmentsAndNavDrawer/app/src/main/java/com/example/fragmentsandnavdrawer/MainActivity.kt
package com.example.fragmentsandnavdrawer

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.viewpager2.widget.ViewPager2
import com.example.fragmentsandnavdrawer.adapter.ViewPagerAdapter
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator

class MainActivity : AppCompatActivity() {


    val arrayTabs = arrayOf("Monday","Tuesday","Wednesday")


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val viewpager:ViewPager2 = findViewById(R.id.viewpager2)
        val tabLayout:TabLayout = findViewById(R.id.tabLayout)

        val viewPagerAdapter = ViewPagerAdapter(supportFragmentManager,lifecycle)
        viewpager.adapter = viewPagerAdapter

        TabLayoutMediator(tabLayout,viewpager){
            tab, position -> tab.text = arrayTabs[position]
        }.attach()


    }
}